### PR TITLE
Implemented support for Bitbucket

### PR DESCRIPTION
**Pull request by swenz**
I implemented support for bitbucket oauth authentication, as a template I used the AspNet.Security.OAuth.GitHub project.

### CODE REVIEW

**Comment by PinpointTownes** [comment]
Thanks for your contribution :+1: 

I'll take a look and merge it when I have a minute.

**Comment by PinpointTownes** [comment]
Sorry for not getting back to you sooner. Reviewing and merging all the PRs of this repo is on my todo list, but I'm currently overbooked with my professional activities. I'll try to dedicate an hour or two next week.

Thanks.

File: src/AspNet.Security.OAuth.Bitbucket/BitbucketAuthenticationOptions.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            AuthorizationEndpoint = BitbucketAuthenticationDefaults.AuthorizationEndpoint;
+            TokenEndpoint = BitbucketAuthenticationDefaults.TokenEndpoint;
+            UserInformationEndpoint = BitbucketAuthenticationDefaults.UserInformationEndpoint;
+        }
+
+        /// <summary>
+        /// Gets or sets the address of the endpoint exposing
+        /// the email addresses associated with the logged in user.
+        /// </summary>
+        public string UserEmailsEndpoint { get; set; } = BitbucketAuthenticationDefaults.UserEmailsEndpoint;
```
Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.Bitbucket/BitbucketAuthenticationOptions.cs]: So Bitbucket uses the same approach as GitHub and provide a separate endpoint to retrieve the email addresses? Interesting.

Comment (swenz) [inline src/AspNet.Security.OAuth.Bitbucket/BitbucketAuthenticationOptions.cs]: Yes exactly you have to do this second step in order to get the mail address.

**Comment by PinpointTownes** [comment]
Merged, thanks for your contribution :clap: 

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/commit/561be0519378670a56395c21ea0c606e87803170

**Pull request closed without merging**